communication/data_logging.hpp

```diff
@@ -280,15 +299,58 @@ class Data_logging
     State& state_;                              ///< The pointer to the state structure
 };
 
+
+/**
+ * \brief   Log data in files
+ *
+ * \tparam  N   Maximum number of variables to log
+ */
+template<uint32_t N>
+class Data_logging_tpl: public Data_logging
+{
+public:
+    /**
+     * \brief   Data logging constructor
+     */
+    Data_logging_tpl(File& file, State& state, data_logging_conf_t config = data_logging_default_config()):
+        Data_logging(file, state, config)
+    {}
+
+protected:
+
+    /**
+     * \brief  Get maximum number of variables to log
+     *
+     * \return Maximum number of variables to log
+     */
+    uint32_t max_count(void)
+    {
+        return N;
+    }
+
+    /**
+     * \brief  Get list of variables to log
+     *
+     * \return list
+     */
+    data_logging_entry_t* list()
+    {
+        return list_;
+    }
+
+private:
+    data_logging_entry_t list_[N];            ///< Data logging array, needs memory allocation
+};
+
+
 static inline data_logging_conf_t data_logging_default_config()
```

Comment:
done in https://github.com/lis-epfl/MAVRIC_Library/pull/326/commits/6de7ea699913a16a94ae41ba7e0c8cca1cc28555